### PR TITLE
chore: move property animation file

### PR DESCRIPTION
**Pull request by Codeboy-cn**
- add cubic bezier
- add engine pause/resume api
- move property animation file

### CODE REVIEW

File: src/Engine3D.ts [hunk 1]
```diff
@@ -321,4 +321,18 @@ export class Engine3D {
 
         if (this._lateRender) this._lateRender();
     }
+
+    /**
+     * Pause the engine render
+     */
+    public static pause() {
```
Comment (lslzl3000) [inline src/Engine3D.ts:328]: How about cancelAnimationFrame?

File: src/Engine3D.ts
Comment (lslzl3000) [inline src/Engine3D.ts:1]: need to add other public apis comments for Engine3D

File: src/Engine3D.ts [hunk 2]
```diff
@@ -321,4 +321,18 @@ export class Engine3D {
 
         if (this._lateRender) this._lateRender();
```
Comment (hellmor) [inline src/Engine3D.ts:322]: code style.

File: src/math/CubicBezierCurve.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        this.setControlVerts(cvs);
+    }
+
+    /**
+     * update controller points
+     * @param cvs controller points
+     */
+    public setControlVerts(cvs: Vector3[]) {
+        // Cubic Bezier curves require 4 cvs.
+        if (cvs.length == 4) this.controlVerts = cvs.concat();
```
Comment (hellmor) [inline src/math/CubicBezierCurve.ts:24]: code style.

**Review by SirXuNian — APPROVED** [review]

**Review by SirXuNian — APPROVED** [review]

**Review by hellmor — APPROVED** [review]

**Pull request merged**